drivers/media/i2c/tc358743.c

```diff
@@ -1458,28 +1458,29 @@ static int tc358743_g_mbus_config(struct v4l2_subdev *sd,
 			     struct v4l2_mbus_config *cfg)
 {
 	struct tc358743_state *state = to_state(sd);
+	const u32 mask = V4L2_MBUS_CSI2_LANE_MASK;
+
+	if (state->csi_lanes_in_use > state->bus.num_data_lanes)
+		return -EINVAL;
 
 	cfg->type = V4L2_MBUS_CSI2;
+	cfg->flags = (state->csi_lanes_in_use << __ffs(mask)) & mask;
 
-	/* Support for non-continuous CSI-2 clock is missing in the driver */
-	cfg->flags = V4L2_MBUS_CSI2_CONTINUOUS_CLOCK;
+	/* In DT mode, only report the number of active lanes */
+	if (sd->dev->of_node)
+		return 0;
 
-	switch (state->csi_lanes_in_use) {
-	case 1:
+	/* Support for non-continuous CSI-2 clock is missing in pdata mode */
+	cfg->flags |= V4L2_MBUS_CSI2_CONTINUOUS_CLOCK;
```

Comment:
Why is this assignment unconditional, and not copying (some of) `state->bus.flags`?

Reply:
Sorry, missed this one earlier. That's part of an upstream commit by Philipp Zabel that needs to be pinged. https://patchwork.linuxtv.org/patch/44397/

Much discussion on the linux-media over g_mbus_config fullstop. Hans Verkuil was fighting that it was unneeded and deprecated as DT conveys all the info, but that doesn't handle the case where the device (eg TC358743) can dynamically choose to use fewer lanes than are physically present (DT only describes those physically present).
This line is part of the original mode of operation, but is now only valid for platform data driven systems (Cisco use this chip and driver in one of their video conferencing products via platform data). DT systems have already aborted by this point.